sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/io/BigQueryIO.java

```diff
@@ -961,7 +970,25 @@ public PDone apply(PCollection<TableRow> input) {
           return input.apply(new StreamWithDeDup(table, tableRefFunction, schema));
         }
 
-        return PDone.in(input.getPipeline());
+        String tempLocation = options.getTempLocation();
+        Preconditions.checkArgument(!Strings.isNullOrEmpty(tempLocation));
+        String tempFilePrefix = tempLocation + "/BigQuerySinkTemp";
```

Comment:
Shouldn't there be some random ID in this path? What if user writes to multiple BQ sinks in the same job?